core/src/main/java/org/wso2/testgrid/core/command/CleanUpCommand.java

```diff
@@ -232,6 +252,19 @@ private void clearDataSources(String datasource) {
         }
     }
 
+    /**
+     * This method delete the files in s3 for a given test plan
+     * @param testPlanId tes plan id of which the files need to be deleted
+     * @throws TestGridDAOException
+     */
+    public void deleteS3(String testPlanId) throws TestGridDAOException {
+
+        Optional<TestPlan> testPlanEntity = testPlanUOW.getTestPlanById(testPlanId);
+        TestPlan testPlan = testPlanEntity.get();
```

Comment:
Possible cause for NPE. In case the test-plan got deleted from the db by some other means, this call will fail. We need to pay attention concurrency situations.